internal/storage/postgres/tree/get_test.go

```
@@ -0,0 +1,534 @@
+		// given
+		suite.ResetDB(t)
+		r := NewTreeRepository(suite.Store, mappers)
+		latitude := 0.0
+		longitude := 0.0
+
+		// when
+		tree, err := r.GetByCoordinates(context.Background(), latitude, longitude)
+
+		// then
+		assert.Error(t, err)
+		assert.Nil(t, tree)
+	})
+
+	t.Run("should return error when context is canceled", func(t *testing.T) {
+		// given
+		r := NewTreeRepository(suite.Store, mappers)
+		ctx, cancel := context.WithCancel(context.Background())
+		cancel()
+
+		// when
+		tree, err := r.GetByCoordinates(ctx, 54.821248093376, 9.485710628517)
+
+		// then
+		assert.Error(t, err)
+		assert.Nil(t, tree)
+	})
+}
+
+func TestTreeRepository_GetAllImagesByID(t *testing.T) {
+	t.Run("should return all images for the given tree ID", func(t *testing.T) {
+		// given
+		suite.ResetDB(t)
+		suite.InsertSeed(t, "internal/storage/postgres/seed/test/tree")
+		r := NewTreeRepository(suite.Store, mappers)
+		treeID := int32(1)
+
+		// when
+		images, err := r.GetAllImagesByID(context.Background(), treeID)
+
+		// then
+		assert.NoError(t, err)
+		assert.NotNil(t, images)
+		assert.NotEmpty(t, images, "Images list should not be empty")
+		for _, image := range images {
+			assert.NotZero(t, image.ID, "Image ID should not be zero")
+			assert.NotEmpty(t, image.URL, "Image URL should not be empty")
+			assert.NotEmpty(t, image.Filename, "Image Filename should not be empty")
+			assert.NotEmpty(t, image.MimeType, "Image MIME type should not be empty")
+		}
+	})
+
+	t.Run("should return an empty list when the tree has no images", func(t *testing.T) {
+		// given
+		suite.ResetDB(t)
+		r := NewTreeRepository(suite.Store, mappers)
+		treeID := int32(2)
+
+		// when
+		images, err := r.GetAllImagesByID(context.Background(), treeID)
+
+		// then
+		assert.NoError(t, err)
+		assert.NotNil(t, images)
+		assert.Empty(t, images, "Images list should be empty")
+	})
+
+	t.Run("should return an empty list when tree is not found", func(t *testing.T) {
```

I think this should return an error and not be successful.